server/bound_queries.cc

```diff
@@ -935,16 +935,16 @@ BoundSelectQuery(const SelectExpression & select,
 {
     try {
 
-        ExcAssert(where);
+        ExcAssert(&where);
```

Comment:
It catches errors in the upstream components.  Agree we should remove it once it hasn't triggered for a while.